File: Selenium/Project/ProjectLMS21Jun.java
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

public class ProjectLMS21Jun {

    WebDriver driver;
    String username = "root";
    String password = "pa$$w0rd";

    @BeforeClass
    public void beforeMethod() {
        // Set up the Firefox driver
        WebDriverManager.firefoxdriver().setup();
        //Create a new instance of the Firefox driver
        driver = new FirefoxDriver();

        //Open browser
        driver.get("https://alchemy.hguy.co/lms");
    }

    @Test
    public void getTitle() {
        String title = driver.getTitle();

        //Print the title of the page
        System.out.println("Page title is: " + title);
        Assert.assertEquals(title, "Alchemy LMS – An LMS Application");

    }

    @Test
    public void getHeading() {
        String heading = driver.findElement(By.className("uagb-ifb-title")).getText();

        //Print the title of the page
        System.out.println("Page Heading is: " + heading);
        Assert.assertEquals(heading, "Learn from Industry Experts");
    }

    @Test
    public void firstInfoBox() {
        String text = driver.findElement(By.xpath("//h3[text()=' Actionable Training ']")).getText();
        System.out.println("Page text in first info box is: " + text);
        Assert.assertEquals(text, "Actionable Training");
    }

    @Test
    public void secondMostPopularCourse() {
        String text1 = driver.findElement(By.xpath("//h3[text()='Email Marketing Strategies']")).getText();
        System.out.println("Page text in first info box is: " + text1);
        Assert.assertEquals(text1, "Email Marketing Strategies");

    }

    @Test
    public void navigateToAnotherPage() {
        driver.findElement(By.xpath("//*[text()='My Account']")).click();
        String myAcctTitle = driver.getTitle();
        System.out.println("My Account Page title is: " + myAcctTitle);
        String myAcct = driver.findElement(By.xpath("//h1[text()='  My Account    ']")).getText();
        System.out.println("My Account Page heading is: " + myAcct);
        Assert.assertEquals(myAcct, "My Account");

    }

    @Test
    public void login() {
        navigateToAnotherPage();
        driver.findElement(By.xpath("//*[contains(@class,'ld-login ld-login ld-login-text ld-login-button ld-button')]")).click();
        driver.findElement(By.id("user_login")).sendKeys(username);
        driver.findElement(By.id("user_pass")).sendKeys(password);
        driver.findElement(By.id("wp-submit")).click();
        String loggedInPageTitle = driver.getTitle();
        System.out.println("Logged in page title is: " + loggedInPageTitle);
        String user = driver.findElement(By.xpath("/html/body/div[2]/div/ul[2]/li[2]/a/span")).getText();
        System.out.println("Logged in user is: " + user);
        Assert.assertEquals(user, "root");
    }

    @Test
    public void countNoOfCourses() {
        driver.findElement(By.xpath("/html/body/div[1]/header/div/div/div/div/div[3]/div/nav/div/ul/li[2]/a")).click();
        String titleAllCourses = driver.getTitle();
        System.out.println("All courses page title is: " + titleAllCourses);
        Assert.assertEquals(driver.getTitle(), "All Courses – Alchemy LMS");
        int mycount = driver.findElements(By.xpath("//div[contains(@class, 'ld_course_grid col-sm-8 col-md-4')]")).size();
        System.out.println("Count of the course is : " + mycount);
    }

    @Test
    public void contactAdmin() {
        driver.findElement(By.xpath("//*[contains(@href,'https://alchemy.hguy.co/lms/contact/')]")).click();
        System.out.println("you have reach contact-us page");
        driver.findElement(By.id("wpforms-8-field_0")).sendKeys("Shashank Shanker");
        driver.findElement(By.id("wpforms-8-field_1")).sendKeys("shashank.shanker@ibm.com");
        driver.findElement(By.id("wpforms-8-field_3")).sendKeys("Help Required with courses...");
        driver.findElement(By.id("wpforms-8-field_2")).sendKeys("How are you doing today");
        driver.findElement(By.id("wpforms-submit-8")).click();
        System.out.println("you have successfully submitted query");
        String message = driver.findElement(By.xpath("/html/body/div[1]/div/div/div/main/article/div/section[4]/div[2]/div[2]/div[2]/div[2]/p")).getText();
        System.out.println("After submission, this message has been displayed:" + message);
    }

    @Test
    public void completeSimpleLesson() {
        login();
        countNoOfCourses();
        driver.findElement(By.xpath("//*[contains(@href,'https://alchemy.hguy.co/lms/courses/social-media-marketing/')]")).click();
        
    }


    @AfterClass
    public void closeBrowser(){
        driver.quit();
    }
}

File: Selenium/Project/ProjectLMS21Jun.py
import time
from typing import Self
import unittest
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.common.action_chains import ActionChains 
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.ui import Select
# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

username = "root"
password = "pa$$w0rd"
# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Initialize the wait object
    wait = WebDriverWait(driver, 10)

    # Navigate to the URL
    driver.get("https://alchemy.hguy.co/lms")
    # Print the title of the page
    pagetitle = driver.title
    assert driver.title == "Alchemy LMS – An LMS Application"
    print("Page title is: ", pagetitle)

    heading = driver.find_element(By.XPATH, "//h1[@class='uagb-ifb-title']").text
    print("Page heading is: ", heading)
    assert heading == "Learn from Industry Experts"

    titlefirstbox = driver.find_element(By.XPATH, "//h3[@class='uagb-ifb-title']").text
    print("Title of first info box is: ", titlefirstbox)
    assert titlefirstbox == "Actionable Training"

    titlesecondcourse = driver.find_element(By.XPATH, "//h3[text()='Email Marketing Strategies']").text
    print("Title of second most popular course is: ", titlesecondcourse)
    assert titlesecondcourse == "Email Marketing Strategies"

    driver.find_element(By.XPATH, "//a[text()='My Account']").click()
    myaccountpagetitle = driver.title
    assert myaccountpagetitle == "My Account – Alchemy LMS"
    print("You are now on My Account page and title of the page is:" + myaccountpagetitle)

    driver.find_element(By.XPATH, "//*[contains(@class,'ld-login ld-login ld-login-text ld-login-button ld-button')]").click()
    driver.find_element(By.ID,'user_login').send_keys(username)
    driver.find_element(By.ID,'user_pass').send_keys(password)
    driver.find_element(By.ID,'wp-submit').click()

    
    user = driver.find_element(By.XPATH,"//*[contains(@class,'display-name')]").text
    print(user + " user is logged in")

    driver.find_element(By.XPATH, "/html/body/div[1]/header/div/div/div/div/div[3]/div/nav/div/ul/li[2]/a").click()
    titleAllCourses = driver.title
    print(titleAllCourses + " is the page title now")
    assert titleAllCourses == "All Courses – Alchemy LMS"

    count = len(driver.find_elements(By.XPATH,"//div[contains(@class, 'ld_course_grid col-sm-8 col-md-4')]"))
    print (str(count) + " courses are present")

    driver.find_element(By.XPATH, "//*[contains(@href,'https://alchemy.hguy.co/lms/contact/')]").click()
    print ("you have reached contact-us page")
    driver.find_element(By.ID , 'wpforms-8-field_0').send_keys("Shashank Shanker")
    driver.find_element(By.ID , 'wpforms-8-field_1').send_keys("shashank.shanker@ibm.com")
    driver.find_element(By.ID , 'wpforms-8-field_3').send_keys("Help Required with courses...")
    driver.find_element(By.ID, 'wpforms-8-field_2').send_keys("How are you doing today")
    driver.find_element(By.ID, 'wpforms-submit-8').click()
    print ("you have successfully submitted query")
    message = driver.find_element(By.XPATH, '/html/body/div[1]/div/div/div/main/article/div/section[4]/div[2]/div[2]/div[2]/div[2]/p').text
    print ("After submission, this message has been displayed : " + message)

    driver.find_element(By.XPATH, "/html/body/div[1]/header/div/div/div/div/div[3]/div/nav/div/ul/li[2]/a").click()
    driver.find_element(By.XPATH, "//*[contains(@class,'btn btn-primary')]").click()
    driver.find_element(By.XPATH, "//div[contains(@class, 'ld-item-title')]").click()
    driver.find_element(By.XPATH, "//input[contains(@class, 'learndash_mark_complete_button')]").click()
    print ("Course completed")